src/libsync/propagatedownloadzsync.cpp

```diff
@@ -154,35 +158,35 @@ void GETFileZsyncJob::seedFinished(void* zs)
         return;
     }
 
+    qCDebug(lcZsyncGet) << "Number of ranges:" << _nrange;
+
     /* If we have no ranges then we have equal files and we are done */
     if (_nrange == 0) {
-        free(_zbyterange);
+        _propagator->reportFileTotal(*_item, 0);
         emit finishedSignal();
         return;
     }
 
-    qCDebug(lcZsyncGet) << "Number of ranges:" << _nrange;
-
-    _zr = zsync_begin_receive(zsp.get(), 0);
+    _zr = zsync_unique_ptr<struct zsync_receiver> (zsync_begin_receive(_zs.get(), 0), [](struct zsync_receiver* zr) {
+       zsync_end_receive(zr);
```

Comment:
Are there any order-of-destruction interactions between _zs, _zr and _zbyteranges? Is it safe to destroy _zs before _zr? This would be decided by the order of declaration in the header and must have a comment if that's something this depends on.

Reply:
Done.